contracts/src/set_verifier.rs

```diff
@@ -116,4 +133,110 @@ where
 
         Ok((image_id, image_url))
     }
+
+    /// Returns the seal if of the given verified root.
+    pub async fn get_verified_root_seal(&self, root: B256) -> Result<Bytes> {
```

Comment:
I like the word `query` here to indicate a network call will occur. I think `fetch` is also good.
```suggestion
    pub async fn query_verified_root_seal(&self, root: B256) -> Result<Bytes> {
```